### PR TITLE
Dataloader improvements: option to disable cube maintainence, CubeGen…

### PR DESCRIPTION
**Pull request by twdsilva**
…erator to generate data with non uniform distribution

Adds an option to disable cube maintenance so that the DataLoader can write data to kudu much faster. Adds an initialize method to the generator so that we don't need to synchronize the getData method for some generators. Adds a new cube generator to generate data using a non uniform distribution so that the cube tables that are populate see realistic reduction in the number of rows.  

**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [ ] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

File: adapter/src/main/java/com/twilio/kudu/dataloader/generator/ColumnValueGenerator.java [hunk 1]
```diff
@@ -24,4 +24,9 @@
 
   public abstract Object getColumnValue(String columnName);
 
+  // needed because while reading from json the default constructor is used
+  // without member
+  // variables being initialized first
+  public abstract void initialize();
```
Comment (sdreynolds) [inline adapter/src/main/java/com/twilio/kudu/dataloader/generator/ColumnValueGenerator.java:30]: Probably should be Javadoc formatted
```suggestion
  /** 
    * Initialize the {@code ColumnValueGenerator} from JSON or other files. Used as a alternative constructor
    */
  public abstract void initialize();
```

File: adapter/src/main/java/com/twilio/kudu/dataloader/generator/TimestampGenerator.java [hunk 2]
```diff
@@ -28,16 +28,20 @@ private TimestampGenerator() {
   }
 
   public TimestampGenerator(final long numDaysBefore) {
-    super(System.currentTimeMillis() - numDaysBefore * DateTimeUtils.MILLIS_PER_DAY, System.currentTimeMillis());
+    this.numDaysBefore = numDaysBefore;
```
Comment (sdreynolds) [inline adapter/src/main/java/com/twilio/kudu/dataloader/generator/TimestampGenerator.java:31]: Can you explain this change, it looks weird.

File: adapter/src/main/java/com/twilio/kudu/dataloader/generator/UniformIntegerValueGenerator.java [hunk 3]
```diff
@@ -30,9 +30,15 @@ public UniformIntegerValueGenerator(final int minVal, final int maxVal) {
     this.maxValue = maxVal;
   }
 
+  /**
+   * Generates a int value between [minValue, maxValue)
+   */
   @Override
-  public synchronized Integer getColumnValue() {
+  public Integer getColumnValue() {
```
Comment (sdreynolds) [inline adapter/src/main/java/com/twilio/kudu/dataloader/generator/UniformIntegerValueGenerator.java:37]: Should this be `synchronized` still? Is accessing `random` thread safe? . Is `random` `ThreadLocalRandom`?

**Review by sdreynolds — APPROVED** [review]
This looks fine to me. I left a few suggestions and questions. 

This change uses `System.out.println` -- is it possible to use `slf4j` instead ?

File: adapter/src/main/java/com/twilio/kudu/dataloader/generator/UniformIntegerValueGenerator.java (hunk 3 above)
Comment (twdsilva) [reply adapter/src/main/java/com/twilio/kudu/dataloader/generator/UniformIntegerValueGenerator.java:36]: Good point, modified all the generators to use `ThreadLocalRandom`

File: adapter/src/main/java/com/twilio/kudu/dataloader/generator/TimestampGenerator.java (hunk 2 above)
Comment (twdsilva) [reply adapter/src/main/java/com/twilio/kudu/dataloader/generator/TimestampGenerator.java]: reverted this change

**Comment by twdsilva** [comment]
> This looks fine to me. I left a few suggestions and questions.
> 
> This change uses `System.out.println` -- is it possible to use `slf4j` instead ?

Fone, changed to use logger.

File: adapter/src/test/java/com/twilio/kudu/sql/parser/KuduDDLIT.java [hunk 4]
```diff
@@ -26,6 +26,7 @@
 import org.apache.kudu.client.KuduException;
 import org.apache.kudu.client.KuduTable;
 import org.apache.kudu.client.PartialRow;
+import org.apache.kudu.client.Partition;
```
Comment (sdreynolds) [inline adapter/src/test/java/com/twilio/kudu/sql/parser/KuduDDLIT.java:29]: Looks like an unused import? Spotless only checks the `main` directory for unused imports

Comment (twdsilva) [reply adapter/src/test/java/com/twilio/kudu/sql/parser/KuduDDLIT.java]: removed the unused import

**Pull request merged**